indexing-hadoop/src/main/java/io/druid/indexer/InputRowSerde.java

```diff
@@ -204,6 +213,19 @@ public static final InputRow fromBytes(byte[] data, AggregatorFactory[] aggs)
     }
   }
 
+  private static String getType(String metric, AggregatorFactory[] aggs, int i)
+  {
+    if(aggs[i].getName().equals(metric)) {
+      return aggs[i].getTypeName();
+    }
+    for(AggregatorFactory agg : aggs) {
```

Comment:
please add a log.warn(..) here so that we know if in a real use case this fall backs to for loop.
also it will be nice to have a UT that verifies if ordering of AggregatorFactory[] in serialization and deserialization stayed same then the for loop is not run.